tests/test_permissions.py

```diff
@@ -173,15 +179,16 @@ def test_merge_permissions_denied_process(factory, network):
         spec = factory.create_traintuple(
             algo=algo_2,
             dataset=dataset_1,
-            data_samples=[datasample_1],
+            data_samples=[train_data_sample_1],
         )
         traintuple_2 = session_2.add_traintuple(spec).future().wait()
 
         # failed to add testtuple from node 3
         spec = factory.create_testtuple(
             objective=objective_1,
-            traintuple_spec=traintuple_2,
+            traintuple=traintuple_2,
         )
 
-        with pytest.raises(substra.exceptions.AuthorizationError):
+        with pytest.raises(substra.exceptions.InvalidRequest) as e:
```

Comment:
I'm not sure this is a good change.

The expected exception should be `AuthorizationError` and not `InvalidRequest` here?

Reply:
You're right, and your fix in https://github.com/SubstraFoundation/substra-backend/pull/107 allowed me to revert my change. I also took the liberty to add a very basic permission test checking that you cannot create a tuple referencing assets you don't have the rights to.

Reply:
You're right, and your fix in SubstraFoundation/substra-backend#107 allowed me to revert my change. I also took the liberty to add a very basic permission test checking that you cannot create a tuple referencing assets you don't have the rights to.